units/patch_intent_test.go

```diff
@@ -333,7 +349,14 @@ func (s *PatchIntentUnitsSuite) TestSetToPreviousPatchDefinition() {
 			Name:         "bv_different_dt",
 			DisplayTasks: []patch.DisplayTask{{Name: "dt2"}},
 		},
-	}}
+	},
+		TaskGroups: []model.TaskGroup{
+			{
+				Name:  "tg",
+				Tasks: []string{"tgt1", "tgt2"},
+			},
+		},
+	}
```

Comment:
If a single-host task group task fails, it should supposedly include all the tasks preceding the failed one as well. Does this account for the previous successful tasks?

Can we test the single host task group case?

Reply:
It will only work if you have a dependency on it. 
Would the current behavior be preferable? Or do we just assume that single host tasks are dependent on each other?

Reply:
I think single-host task group dependencies aren't added until the task is about to end ([here](https://github.com/evergreen-ci/evergreen/blob/9e6ebecbc17cef8663460d75703fd3c05af64e63/model/task_queue.go#L279-L283), which is called in the agent end task route), so single-host task groups aren't guaranteed to have the dependencies when the user reuses the patch.

Reply:
updated